### PR TITLE
Implement a status alert banner

### PR DESCRIPTION
**Pull request by BrandonRomano**
This banner conditionally renders whenever we report an issue on https://status.hashicorp.com/.

CleanShot 2020-11-19 at 10 29 22

### CODE REVIEW

File: website/pages/_app.js
Comment (BrandonRomano) [inline website/pages/_app.js:54]: Service alert banner gets precedent over any marketing banners.

**Comment by BrandonRomano** [comment]
Working with some designers to come up with a better presentation here; converting this one to draft as we'll await a better presentation before merging.

**Comment by dizzyup** [comment]
@BrandonRomano Some updates on this -

We're going to show an indicator in the footer to link to status, and then when there's an ongoing service issue we'll show a banner that pins to the bottom of the viewport. In both cases those'll link off to status.hashicorp.com

#### When things are fine...
CleanShot 2020-11-16 at 19 08 33@2x

#### When things are not fine...
CleanShot 2020-11-16 at 13 26 13@2x
CleanShot 2020-11-16 at 13 26 18@2x

File: website/pages/home/HomePage.module.css
Comment (BrandonRomano) [inline website/pages/home/HomePage.module.css:2]: This breaks the terminal animation on the homepage

**Review by BrandonRomano — COMMENTED** [review]
Code generally looks good -- there's an issue with using overflow: hidden with how the terminal behaves.

LMK if you want to kick this to me and I can fix -- otherwise I'll leave it to you to patch up!

Thanks for pushing these changes up!  Looks really good

File: website/pages/home/HomePage.module.css
Comment (dizzyup) [reply website/pages/home/HomePage.module.css]: Argh, good catch - can't believe I missed that.

**Comment by dizzyup** [comment]
> Code generally looks good -- there's an issue with using overflow: hidden with how the terminal behaves.
> 
> LMK if you want to kick this to me and I can fix -- otherwise I'll leave it to you to patch up!
> 
> Thanks for pushing these changes up! Looks really good

Did a bit more digging but couldn't get anywhere - I think this might just be a Safari bug, there's no overflow in Chrome. Not sure I'll be able to get anywhere with it in the near term - seems like it'd need a layout refactor which is probably a bit uncalled for as it really is a minor Safari behaviour thing! If you have any thoughts feel free to take a crack at it!

Otherwise, I've removed that commit and this should be good to go - does anyone else need to give sign-off on this PR?

**Comment by BrandonRomano** [comment]
I'll give this one more sweep a bit later today and take it for a spin.  A bit busy today, so there is some risk it slips to tomorrow, but I have it on my list!

We can share the preview with the broader Waypoint team to ensure they like the behaviors (if you haven't already), beyond that no further approvals are needed and we can ship.

**Review by BrandonRomano — COMMENTED** [review]
Can't approve as I pushed up this PR originally, but this is great work @dizzyup!  Really like the subtle visual touches for hover states.

Sorry for the delay in review here, wanted to take this for a spin locally to force the error state and do some cross browser testing.  All is working great 🍏

**Review by kylemac — APPROVED** [review]

**Pull request merged**
